2006-10-16  Ulrich Drepper  <drepper@redhat.com>

	[BZ #3369]
	* sysdeps/posix/getaddrinfo.c (rfc3484_sort): Fix typos in rules 4
	and 7.

--- libc/sysdeps/posix/getaddrinfo.c	2 Oct 2006 16:49:47 -0000	1.107
+++ libc/sysdeps/posix/getaddrinfo.c	16 Oct 2006 15:19:17 -0000	1.108
@@ -1395,10 +1395,10 @@ rfc3484_sort (const void *p1, const void
     {
       if (!(a1->source_addr_flags & in6ai_homeaddress)
 	  && (a2->source_addr_flags & in6ai_homeaddress))
-	return -1;
+	return 1;
       if ((a1->source_addr_flags & in6ai_homeaddress)
 	  && !(a2->source_addr_flags & in6ai_homeaddress))
-	return 1;
+	return -1;
     }
 
   /* Rule 5: Prefer matching label.  */
@@ -1435,11 +1435,11 @@ rfc3484_sort (const void *p1, const void
   if (a1->got_source_addr)
     {
       if (!(a1->source_addr_flags & in6ai_temporary)
-	  && (a1->source_addr_flags & in6ai_temporary))
+	  && (a2->source_addr_flags & in6ai_temporary))
 	return -1;
       if ((a1->source_addr_flags & in6ai_temporary)
-	  && !(a1->source_addr_flags & in6ai_temporary))
-	return -1;
+	  && !(a2->source_addr_flags & in6ai_temporary))
+	return 1;
 
       /* XXX Do we need to check anything beside temporary addresses?  */
     }
